docs/reference/function-configuration/code-entry-types.md

```diff
@@ -140,23 +141,58 @@ Additional information for performing the download &mdash; such as the download
 > - The downloaded code files are saved and can be used by the function handler.
 
-> **Dashboard Note:** To configure an external function-code source from the dashboard, select the relevant code-entry type &mdash; `Archive`, `GitHub`, or `S3` &mdash; from the **Code entry type** list.
+> **Dashboard Note:** To configure an external function-code source from the dashboard, select the relevant code-entry type &mdash; `Archive`, `Git`, `GitHub`, or `S3` &mdash; from the **Code entry type** list.
 
 The downloaded function code can optionally contain a **function.yaml** file with function configuration for enriching the original configuration (in the configuration file that sets the code-entry type) according to the following merge strategy:
 
 - Field values that are set only in the downloaded configuration are added to the original configuration.
 - List and map field values &mdash; such as `meta.labels` and `spec.env` &mdash;are merged by adding any values that are set only in the downloaded configuration to the values that are set in the original configuration.
 - In case of a conflict &mdash; i.e., if the original and downloaded configurations set different values for the same element &mdash; the original configuration takes precedence and the value in the downloaded configuration is ignored.
 
+<a id="code-entry-type-git"></a>
+### Git code-entry type (`git`)
+
+Set the [`spec.build.codeEntryType`](/docs/reference/function-configuration.md#spec.build.codeEntryType) function-configuration field to `git` (dashboard: **Code entry type** = `Git`) to download the function code from a Git repository. The following configuration fields provide additional information for performing the download:
+
+- `spec.build` &mdash;
+  - `path` (dashboard: **URL**) (Required) &mdash; the URL of the Git repository that contains the function code.
+  - `codeEntryAttributes` &mdash;
+      - `branch` (dashboard: **Branch**) (Required) &mdash; the Git repository branch from which to download the function code.
+      - `gitCredentials` (Optional) &mdash; Git credentials for the download, when required.
+        - `accessToken` (dashboard: **Token**) (Optional) Git access key
+        - `username` (dashboard: **Username**) (Optional) Git username
+        - `password` (dashboard: **Password**) (Optional) Git password
+      - `workDir` (dashboard: **Work directory**) (Optional) &mdash; the relative path to the function-code directory within the configured repository branch.
+      The default work directory is the root directory of the Git repository (`"/"`).
+
+<a id="code-entry-type-git-example"></a>
+#### Example
+
+```yaml
+spec:
+  description: my Go function
+  handler: main:Handler
+  runtime: golang
+  build:
+    codeEntryType: "git"
+    path: "https://git.com/my-organization/my-repository.git"
+    codeEntryAttributes:
+      branch: "my-branch"
+      gitCredentials:
+        accessKey: "myaccesskey"
```

Comment:
imo, provide examples for some of the common repositories
`AzureDevops`
`Github`
`BitBucket`

it can be very helpful. feel free to add a specific CET reference and nest it under `codeEntryType` dir.
there we can add more custom examples for ease and explain its usage to the user.
this feature it going to be used by a lot of people, so more examples == less issues :)